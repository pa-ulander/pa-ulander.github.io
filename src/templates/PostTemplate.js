import React from 'react'
import { graphql, Link } from 'gatsby'
import { MDXRenderer } from 'gatsby-plugin-mdx'
import Layout from '../components/Layout'

const PostTemplate = ({ data, pageContext }) => {
  console.log('data', data)
  console.log('pageContext', pageContext)
  const { frontmatter, body } = data.mdx
  const { previous, next } = pageContext
  return (
    <Layout>
      <h1 className="post-heading">{frontmatter.title}</h1>
      <p className="post-date">{frontmatter.date}</p>
      <article className="post-body">
        <MDXRenderer>{body}</MDXRenderer>
      </article>
      {previous === false ? null : (
        <>
          {previous && (
            <Link to={`${previous.frontmatter.path}`}>
              <button className="previous-next-button">
                {previous.frontmatter.title}
              </button>
            </Link>
          )}
        </>
      )}
      {next === false ? null : (
        <>
          {next && (
            <Link to={`${next.frontmatter.path}`}>
              <button className="previous-next-button">
                {next.frontmatter.title}
              </button>
            </Link>
          )}
        </>
      )}
    </Layout>
  )
}

export default PostTemplate

export const query = graphql`
  query PostBySlug($slug: String!) {
    mdx(frontmatter: { path: { eq: $slug } }) {
      body
      frontmatter {
        title
        date(formatString: "Do MMMM YYYY")
      }
    }
  }
`
