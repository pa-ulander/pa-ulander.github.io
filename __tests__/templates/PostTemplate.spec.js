import React from 'react'
import renderer from 'react-test-renderer'
import { StaticQuery } from 'gatsby'
import PostTemplate from '../../src/templates/PostTemplate'

jest.mock('gatsby-plugin-mdx', () => {
  return {
    MDXRenderer: ({ children }) => {
      return <div>{children}</div>
    },
  }
})

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) => {
    render({
      mdx: {
        body: '',
        frontmatter: {},
      },
    })
  })
})

describe('Post Template', () => {
  it('should render a blog post with mdx', () => {
    const data = {
      mdx: {
        body:
          'function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }\n\nfunction _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }\n\nfunction _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }\n\n/* @jsxRuntime classic */\n\n/* @jsx mdx */\nvar _frontmatter = {\n  "title": "Fourth post has bears",\n  "description": "Fourth post meta description",\n  "date": "2020-12-04T00:00:00.000Z",\n  "featuredImage": "./image.jpeg",\n  "featuredText": "Fourth post featured text",\n  "pageType": "blog",\n  "templateKey": "blog-post",\n  "published": true,\n  "slug": "fourth-post-has-bears",\n  "path": "/blog/fourth-post-has-bears",\n  "tags": ["dogs", "post"]\n};\nvar layoutProps = {\n  _frontmatter: _frontmatter\n};\nvar MDXLayout = "wrapper";\nreturn function MDXContent(_ref) {\n  var components = _ref.components,\n      props = _objectWithoutProperties(_ref, ["components"]);\n\n  return mdx(MDXLayout, _extends({}, layoutProps, props, {\n    components: components,\n    mdxType: "MDXLayout"\n  }), mdx("p", null, mdx("span", _extends({\n    parentName: "p"\n  }, {\n    "className": "gatsby-resp-image-wrapper",\n    "style": {\n      "position": "relative",\n      "display": "block",\n      "marginLeft": "auto",\n      "marginRight": "auto",\n      "maxWidth": "1500px"\n    }\n  }), "\\n      ", mdx("a", _extends({\n    parentName: "span"\n  }, {\n    "className": "gatsby-resp-image-link",\n    "href": "/static/b10a2b12966b76e153cbed58818309a3/c58a3/image.jpg",\n    "style": {\n      "display": "block"\n    },\n    "target": "_blank",\n    "rel": "noopener"\n  }), "\\n    ", mdx("span", _extends({\n    parentName: "a"\n  }, {\n    "className": "gatsby-resp-image-background-image",\n    "style": {\n      "paddingBottom": "33.3984375%",\n      "position": "relative",\n      "bottom": "0",\n      "left": "0",\n      "backgroundImage": "url(\'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAHABQDASIAAhEBAxEB/8QAFwABAAMAAAAAAAAAAAAAAAAAAAEDBP/EABYBAQEBAAAAAAAAAAAAAAAAAAEAA//aAAwDAQACEAMQAAABo0mNID//xAAZEAACAwEAAAAAAAAAAAAAAAAAAgEDFBL/2gAIAQEAAQUCS+e9Lmhz/8QAFhEBAQEAAAAAAAAAAAAAAAAAAQAR/9oACAEDAQE/AVbb/8QAGBEAAgMAAAAAAAAAAAAAAAAAABEBEjH/2gAIAQIBAT8BVcHJ/8QAGhAAAAcAAAAAAAAAAAAAAAAAAAEQETFBUf/aAAgBAQAGPwIqbEkf/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEhQWGh/9oACAEBAAE/IZulYNOGw//aAAwDAQACAAMAAAAQcC//xAAXEQEAAwAAAAAAAAAAAAAAAAAAAREx/9oACAEDAQE/EJWrP//EABcRAAMBAAAAAAAAAAAAAAAAAAABUWH/2gAIAQIBAT8QSQND/8QAGRABAAMBAQAAAAAAAAAAAAAAAQARUUFh/9oACAEBAAE/EBoKE09l6vZVUCYBP//Z\')",\n      "backgroundSize": "cover",\n      "display": "block"\n    }\n  })), "\\n  ", mdx("img", _extends({\n    parentName: "a"\n  }, {\n    "className": "gatsby-resp-image-image",\n    "alt": "image info",\n    "title": "image info",\n    "src": "/static/b10a2b12966b76e153cbed58818309a3/c58a3/image.jpg",\n    "srcSet": ["/static/b10a2b12966b76e153cbed58818309a3/36dd4/image.jpg 512w", "/static/b10a2b12966b76e153cbed58818309a3/72e01/image.jpg 1024w", "/static/b10a2b12966b76e153cbed58818309a3/c58a3/image.jpg 1500w"],\n    "sizes": "(max-width: 1500px) 100vw, 1500px",\n    "style": {\n      "width": "100%",\n      "height": "100%",\n      "margin": "0",\n      "verticalAlign": "middle",\n      "position": "absolute",\n      "top": "0",\n      "left": "0"\n    },\n    "loading": "lazy"\n  })), "\\n  "), "\\n    ")), mdx("h2", null, "Lorem ipsum"), mdx("p", null, "Laborum nisi amet incididunt officia occaecat ullamco sint occaecat qui. Anim et culpa qui pariatur quis sit reprehenderit ipsum amet commodo cupidatat eiusmod deserunt mollit. Anim nostrud et fugiat eiusmod nisi cillum Lorem incididunt sit est minim dolore."), mdx("h3", null, "A beary list"), mdx("ol", null, mdx("li", {\n    parentName: "ol"\n  }, "one bear"), mdx("li", {\n    parentName: "ol"\n  }, "two bears"), mdx("li", {\n    parentName: "ol"\n  }, "three bears"), mdx("li", {\n    parentName: "ol"\n  }, "four bears")));\n}\n;\nMDXContent.isMDXComponent = true;',
        frontmatter: {
          title: 'Fourth post has bears',
          date: '4th December 2020',
          description: 'Fourth post meta description',
          templateKey: 'blog-post',
          featuredText: 'Fourth post featured text',
        },
      },
    }

    const tree = renderer
      .create(<PostTemplate data={data} pageContext={{}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
