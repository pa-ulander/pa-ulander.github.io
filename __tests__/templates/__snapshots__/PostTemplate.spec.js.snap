// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Template should render a blog post with mdx 1`] = `
Array [
  <div
    className="nav"
  >
    <ul>
      <li>
        <a
          className="navButton navButton__active"
          href="/"
        >
          HOME
        </a>
      </li>
      <li>
        <a
          className="navButton null"
          href="/blog"
        >
          BLOG
        </a>
      </li>
      <li>
        <a
          className="navButton null"
          href="/cv"
        >
          CV
        </a>
      </li>
      <li
        className="toggler"
      >
        <div
          className="dark-button"
        >
          <input
            checked={true}
            id="toggle"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            htmlFor="toggle"
          />
        </div>
      </li>
      <li
        className="toggler"
      />
    </ul>
    <svg
      className="navprinticon"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Skriv ut
      </title>
      <path
        clipRule="evenodd"
        d="M8 4H16V6H8V4ZM18 6H22V18H18V22H6V18H2V6H6V2H18V6ZM20 16H18V14H6V16H4V8H20V16ZM8 16H16V20H8V16ZM8 10H6V12H8V10Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </div>,
  <div
    className="container"
  >
    <h1
      className="post-heading"
    >
      Fourth post has bears
    </h1>
    <p
      className="post-date"
    >
      4th December 2020
    </p>
    <article
      className="post-body"
    >
      <div>
        function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i &lt; arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i &lt; sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) &gt;= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i &lt; sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) &gt;= 0) continue; target[key] = source[key]; } return target; }

/* @jsxRuntime classic */

/* @jsx mdx */
var _frontmatter = {
  "title": "Fourth post has bears",
  "description": "Fourth post meta description",
  "date": "2020-12-04T00:00:00.000Z",
  "featuredImage": "./image.jpeg",
  "featuredText": "Fourth post featured text",
  "pageType": "blog",
  "templateKey": "blog-post",
  "published": true,
  "slug": "fourth-post-has-bears",
  "path": "/blog/fourth-post-has-bears",
  "tags": ["dogs", "post"]
};
var layoutProps = {
  _frontmatter: _frontmatter
};
var MDXLayout = "wrapper";
return function MDXContent(_ref) {
  var components = _ref.components,
      props = _objectWithoutProperties(_ref, ["components"]);

  return mdx(MDXLayout, _extends({}, layoutProps, props, {
    components: components,
    mdxType: "MDXLayout"
  }), mdx("p", null, mdx("span", _extends({
    parentName: "p"
  }, {
    "className": "gatsby-resp-image-wrapper",
    "style": {
      "position": "relative",
      "display": "block",
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "1500px"
    }
  }), "\\n      ", mdx("a", _extends({
    parentName: "span"
  }, {
    "className": "gatsby-resp-image-link",
    "href": "/static/b10a2b12966b76e153cbed58818309a3/c58a3/image.jpg",
    "style": {
      "display": "block"
    },
    "target": "_blank",
    "rel": "noopener"
  }), "\\n    ", mdx("span", _extends({
    parentName: "a"
  }, {
    "className": "gatsby-resp-image-background-image",
    "style": {
      "paddingBottom": "33.3984375%",
      "position": "relative",
      "bottom": "0",
      "left": "0",
      "backgroundImage": "url('data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAHABQDASIAAhEBAxEB/8QAFwABAAMAAAAAAAAAAAAAAAAAAAEDBP/EABYBAQEBAAAAAAAAAAAAAAAAAAEAA//aAAwDAQACEAMQAAABo0mNID//xAAZEAACAwEAAAAAAAAAAAAAAAAAAgEDFBL/2gAIAQEAAQUCS+e9Lmhz/8QAFhEBAQEAAAAAAAAAAAAAAAAAAQAR/9oACAEDAQE/AVbb/8QAGBEAAgMAAAAAAAAAAAAAAAAAABEBEjH/2gAIAQIBAT8BVcHJ/8QAGhAAAAcAAAAAAAAAAAAAAAAAAAEQETFBUf/aAAgBAQAGPwIqbEkf/8QAGRAAAgMBAAAAAAAAAAAAAAAAAAEhQWGh/9oACAEBAAE/IZulYNOGw//aAAwDAQACAAMAAAAQcC//xAAXEQEAAwAAAAAAAAAAAAAAAAAAAREx/9oACAEDAQE/EJWrP//EABcRAAMBAAAAAAAAAAAAAAAAAAABUWH/2gAIAQIBAT8QSQND/8QAGRABAAMBAQAAAAAAAAAAAAAAAQARUUFh/9oACAEBAAE/EBoKE09l6vZVUCYBP//Z')",
      "backgroundSize": "cover",
      "display": "block"
    }
  })), "\\n  ", mdx("img", _extends({
    parentName: "a"
  }, {
    "className": "gatsby-resp-image-image",
    "alt": "image info",
    "title": "image info",
    "src": "/static/b10a2b12966b76e153cbed58818309a3/c58a3/image.jpg",
    "srcSet": ["/static/b10a2b12966b76e153cbed58818309a3/36dd4/image.jpg 512w", "/static/b10a2b12966b76e153cbed58818309a3/72e01/image.jpg 1024w", "/static/b10a2b12966b76e153cbed58818309a3/c58a3/image.jpg 1500w"],
    "sizes": "(max-width: 1500px) 100vw, 1500px",
    "style": {
      "width": "100%",
      "height": "100%",
      "margin": "0",
      "verticalAlign": "middle",
      "position": "absolute",
      "top": "0",
      "left": "0"
    },
    "loading": "lazy"
  })), "\\n  "), "\\n    ")), mdx("h2", null, "Lorem ipsum"), mdx("p", null, "Laborum nisi amet incididunt officia occaecat ullamco sint occaecat qui. Anim et culpa qui pariatur quis sit reprehenderit ipsum amet commodo cupidatat eiusmod deserunt mollit. Anim nostrud et fugiat eiusmod nisi cillum Lorem incididunt sit est minim dolore."), mdx("h3", null, "A beary list"), mdx("ol", null, mdx("li", {
    parentName: "ol"
  }, "one bear"), mdx("li", {
    parentName: "ol"
  }, "two bears"), mdx("li", {
    parentName: "ol"
  }, "three bears"), mdx("li", {
    parentName: "ol"
  }, "four bears")));
}
;
MDXContent.isMDXComponent = true;
      </div>
    </article>
  </div>,
]
`;
